src/csharp/Grpc.Core.Api/ServerCallContext.cs

```diff
@@ -113,6 +116,23 @@ public WriteOptions WriteOptions
         /// </summary>
         public AuthContext AuthContext => AuthContextCore;
 
+        /// <summary>
+        /// Gets a dictionary that can be used by the various interceptors and handlers of this
+        /// call to store arbitrary state.
+        /// </summary>
+        public virtual IDictionary<object, object> UserState
```

Comment:
I'm wondering if we should aim for consistency and follow the pattern of having the properties non-virtual and delegating to UserStateCore property  or if current approach is better.

Reply:
The [Framework Design Guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/virtual-members) suggests the pattern you mention. It will also be more consistent with what we already have. Let me update this.